src/main/kotlin/net/leanix/vsm/gitlab/broker/connector/adapter/graphql/GitlabGraphqlProvider.kt

```diff
@@ -113,7 +138,25 @@ fun Project.toRepository() = Repository(
     languages = languages?.map { Language(it.name, it.name, it.share!!) },
     tags = topics,
     defaultBranch = repository?.rootRef ?: "empty-branch",
-    groupName = group!!.fullPath
+    groupName = group!!.fullPath,
+    path = path
+)
+
+fun MergeRequest.toPullRequest() = PullRequest(
+    id = id,
+    baseRefName = targetBranch,
+    mergedAt = mergedAt.toString(),
+    commits = commits
+        ?.nodes
+        ?.filterNotNull()
+        ?.map {
+            Commit(
+                id = it.id,
+                changeTime = it.authoredDate.toString(),
+                author = Author(it.author?.name ?: "", it.authorEmail ?: "", it.author?.username)
```

Comment:
I'm not sure if we could accept an author with empty email(I think we use the email to link the metrics with the team).
Can you double-check with team helios how we link the metrics with the team?

Reply:
But is there any case the emails can be null?
On Bitbucket, it's allowed. So we're filtering them.

Reply:
done

Reply:
the thing is we don't even allow it on **vsm-events-broker**

```kt
data class Commit(
    val id: String,
    val changeTime: String,
    val author: Author,
)

data class Author(
    val name: String,
    val email: String,
    val username: String?,
)
```